source/server.go

```diff
@@ -106,10 +113,10 @@ func (s *Server) recvRecords(stream pb.SourceService_StreamServer) error {
 }
 
 func (s *Server) SendAck(position sdk.Position) error {
-	stream := s.stream.Load()
-	if stream == nil {
-		return fmt.Errorf("no stream is open")
+	for s.stream.Load() == nil {
+		<-s.StreamIsSetCh
 	}
+	stream := s.stream.Load()
```

Comment:
In the happy path we now load the stream twice. I'd rather turn it around so we only load it once if it's there.
```suggestion
	stream := s.stream.Load()
	for stream == nil {
		<-s.StreamIsSetCh
		stream = s.stream.Load()
	}
```